### PR TITLE
Add builder for whereami

### PR DESCRIPTION
**Pull request by ararslan**
Cross-platform library for identifying where on the filesystem the current executable lives.

### CODE REVIEW

**Comment by giordano** [comment]
I think `-ldl` is needed only with glibc?  And not even with [glibc 2.34 anymore](https://lists.gnu.org/archive/html/info-gnu/2021-08/msg00001.html), but we still use 2.17.

**Comment by ararslan** [comment]
Nice, it appears you are correct, thank you!

File: W/WhereAmI/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,32 @@
+using BinaryBuilder
+
+name = "WhereAmI"
+version = v"2024.8.6"  # no proper versions/releases, this is the date of the commit used
```
Comment (giordano) [inline W/WhereAmI/build_tarballs.jl:4]: For lack of proper veesioning, how about something like 0.1.0 or 1.0.0? Calendar-based veesioning plays very badly with semver

Comment (ararslan) [reply W/WhereAmI/build_tarballs.jl:4]: Sure, makes sense
```suggestion
version = v"1.0.0"  # no proper versions/releases, this is made up
```

**Pull request merged**